subsys/bluetooth/audio/media_proxy.c

```diff
@@ -1706,10 +1706,161 @@ uint8_t media_proxy_ctrl_get_content_ctrl_id(struct media_player *player)
+		BT_DBG("get_track_title is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_track_duration == NULL) {
+		BT_DBG("get_track_duration is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_track_position == NULL) {
+		BT_DBG("get_track_position is NULL");
+		return false;
+	}
+
+	if (pl_calls->set_track_position == NULL) {
+		BT_DBG("set_track_position is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_playback_speed == NULL) {
+		BT_DBG("get_playback_speed is NULL");
+		return false;
+	}
+
+	if (pl_calls->set_playback_speed == NULL) {
+		BT_DBG("set_playback_speed is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_seeking_speed == NULL) {
+		BT_DBG("get_seeking_speed is NULL");
+		return false;
+	}
+
+#ifdef CONFIG_BT_MPL_OBJECTS
+	if (pl_calls->get_track_segments_id == NULL) {
+		BT_DBG("get_track_segments_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_current_track_id == NULL) {
+		BT_DBG("get_current_track_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->set_current_track_id == NULL) {
+		BT_DBG("set_current_track_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_next_track_id == NULL) {
+		BT_DBG("get_next_track_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->set_next_track_id == NULL) {
+		BT_DBG("set_next_track_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_parent_group_id == NULL) {
+		BT_DBG("get_parent_group_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_current_group_id == NULL) {
+		BT_DBG("get_current_group_id is NULL");
+		return false;
+	}
+
+	if (pl_calls->set_current_group_id == NULL) {
+		BT_DBG("set_current_group_id is NULL");
+		return false;
+	}
+
+#endif /* CONFIG_BT_MPL_OBJECTS */
+	if (pl_calls->get_playing_order == NULL) {
+		BT_DBG("get_playing_order is NULL");
+		return false;
+	}
+
+	if (pl_calls->set_playing_order == NULL) {
+		BT_DBG("set_playing_order is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_playing_orders_supported == NULL) {
+		BT_DBG("get_playing_orders_supported is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_media_state == NULL) {
+		BT_DBG("get_media_state is NULL");
+		return false;
+	}
+
+	if (pl_calls->send_command == NULL) {
+		BT_DBG("send_command is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_commands_supported == NULL) {
+		BT_DBG("get_commands_supported is NULL");
+		return false;
+	}
+
+#ifdef CONFIG_BT_MPL_OBJECTS
+	if (pl_calls->send_search == NULL) {
+		BT_DBG("send_search is NULL");
+		return false;
+	}
+
+	if (pl_calls->get_search_results_id == NULL) {
+		BT_DBG("get_search_results_id is NULL");
+		return false;
+	}
+
+#endif /* CONFIG_BT_MPL_OBJECTS */
+	if (pl_calls->get_content_ctrl_id == NULL) {
+		BT_DBG("get_content_ctrl_id is NULL");
+		return false;
+	}
+
+	return true;
+}
+
 int media_proxy_pl_register(struct media_proxy_pl_calls *pl_calls)
 {
-	CHECKIF(pl_calls == NULL) {
-		BT_DBG("NULL calls");
+	CHECKIF(!pl_calls_is_valid(pl_calls)) {
+		BT_DBG("Invalid pl_calls");
```

Comment:
This one is redundant as you have a log for each function pointer already

Reply:
Good point